packages/berry-core/sources/scriptUtils.ts

```diff
@@ -92,6 +93,30 @@ type ExecutePackageScriptOptions = {
 };
 
 export async function executePackageScript(locator: Locator, scriptName: string, args: Array<string>, {cwd, project, stdin, stdout, stderr}: ExecutePackageScriptOptions) {
+  const {manifest, binFolder, env, cwd: realCwd} = await initializePackageEnvironment(locator, project, cwd);
+
+  const script = manifest.scripts.get(scriptName);
+  if (!script)
+    return;
+
+  try {
+    return await execute(script, args, {cwd: realCwd, env, stdin, stdout, stderr});
+  } finally {
+    await xfs.removePromise(binFolder);
+  }
+}
+
+export async function executePackageShellcode(locator: Locator, command: string, args: Array<string>, {cwd, project, stdin, stdout, stderr}: ExecutePackageScriptOptions) {
+  const {binFolder, env, cwd: realCwd} = await initializePackageEnvironment(locator, project, cwd);
+
+  try {
+    return await execute(command, args, {cwd: realCwd, env, stdin, stdout, stderr});
+  } finally {
+    await xfs.removePromise(binFolder);
+  }
+}
+
+export async function initializePackageEnvironment(locator: Locator, project: Project, cwd?: string | undefined) {
```

Comment:
```suggestion
async function initializePackageEnvironment(locator: Locator, project: Project, cwd?: string | undefined) {
```

Let's keep it an implementation detail for now.

Reply:
Also, can you turn `project` and `cwd` into an object argument (similar to the other functions)? I like the [named arguments](https://en.wikipedia.org/wiki/Named_parameter) pattern for arguments that are required mostly to complement the subject (here the locator is the subject of the function, while the project / cwd only are its complements).